plugins/autopeering/local.go

```diff
@@ -0,0 +1,100 @@
+package autopeering
+
+import (
+	"fmt"
+	"net"
+	"strconv"
+
+	"github.com/multiformats/go-multiaddr"
+	"go.etcd.io/bbolt"
+
+	"github.com/gohornet/hornet/core/p2p"
+
+	"github.com/iotaledger/hive.go/autopeering/peer"
+	"github.com/iotaledger/hive.go/autopeering/peer/service"
+	"github.com/iotaledger/hive.go/kvstore/bolt"
+	"github.com/iotaledger/hive.go/logger"
+)
+
+// Local defines the local autopeering peer.
+type Local struct {
+	PeerLocal *peer.Local
+	boltDb    *bbolt.DB
+	peerDb    *peer.DB
+}
+
+func newLocal(seed []byte) *Local {
+	log := logger.NewLogger("Local")
+
+	// let the autopeering discover the IP
+	// TODO: is this really necessary?
+	var peeringIP net.IP
+	bindAddr := deps.NodeConfig.Strings(p2p.CfgP2PBindMultiAddresses)[0]
+	multiAddrBindAddr, err := multiaddr.NewMultiaddr(bindAddr)
+	if err != nil {
+		log.Fatalf("unable to parse bind multi address %s", err)
+		return nil
+	}
+	for _, proto := range multiAddrBindAddr.Protocols() {
+		switch proto.Code {
+		case multiaddr.P_IP4:
+			peeringIP = net.IPv4zero
+		case multiaddr.P_IP6:
```

Comment:
Do you maybe also need the case for `multiaddr.P_DNS` ?

Reply:
Not needed.